xarray/backends/cfgrib_.py

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright 2017-2018 European Centre for Medium-Range Weather Forecasts (ECMWF).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors:
+#   Alessandro Amici - B-Open - https://bopen.eu
+#
+
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+
+from .. import Variable
+from ..core import indexing
+from ..core.utils import Frozen, FrozenOrderedDict
+from .common import AbstractDataStore, BackendArray
+from .file_manager import CachingFileManager
+from .locks import ensure_lock, SerializableLock
+
+# FIXME: Add a dedicated lock just in case, even if ecCodes is supposed to be thread-safe in most
+# circumstances. See: https://confluence.ecmwf.int/display/ECC/Frequently+Asked+Questions
+ECCODES_LOCK = SerializableLock()
+
+
+class CfGribArrayWrapper(BackendArray):
+    def __init__(self, datastore, array):
+        self.datastore = datastore
+        self.shape = array.shape
+        self.dtype = array.dtype
+        self.array = array
+
+    def __getitem__(self, key):
+        return indexing.explicit_indexing_adapter(
+            key, self.shape, indexing.IndexingSupport.BASIC, self._getitem)
```

Comment:
Can you verify which forms of indexing cfgrib actually supports? In your previous commit this was OUTER_1VECTOR. See the docstring on IndexingSupport for details.

Reply:
Now that I checked, I can say that *cfgrib* supports `indexing.IndexingSupport.OUTER`. I don't remember how I ended up declaring OUTER_1VECTOR, and since it sounded wrong, in the last commit intended to play it as safe as possible.

Code updated.